network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/primary_heroguts.html

```diff
@@ -47,8 +47,10 @@ <h1 class="tw-h1-heading">{{ page.hero_headline }}</h1>
     <div class="row cutout-wrapper-row">
       <div class="cutout col-12 col-lg-8 pt-4">
         <h1 class="tw-h1-heading mb-0">{% if root.title %}{{ root.title }}{% elif page.header %}{{ page.header }}{% else %}{{ page.title }}{% endif %}</h1>
-        {% if page.intro %}
-        <p class="tw-body-large mt-3 mb-0">{{ page.intro }}</p>
+        {% if page.use_intro_from_parent and root.specific.intro %}
+          <p class="tw-body-large mt-3 mb-0">{{ root.specific.intro }}</p>
+        {% elif page.intro %}
+          <p class="tw-body-large mt-3 mb-0">{{ page.intro }}</p>
```

Comment:
What do you think about moving this to a method on the page and adding some tests for it? There are quite a few cases to test and I think it would be easier to test all these path automatically rather than do it in the template.

Reply:
Excellent suggestion! I'll work on this quickly today. I noticed the campaign pages don't have their own testing setup, should this test be on the primary page?

Reply:
I don't think this should be affecting the primary page 🤔 You might need to create the test setup. You can might be able to use the primary page test setup as an example (maybe even share setup?)

Reply:
I've revised this a bit now, for you to look at again :)